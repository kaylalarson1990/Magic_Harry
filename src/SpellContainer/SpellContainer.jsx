import React, { Component } from "react";
import { connect } from "react-redux";
import "../HouseContainer/Container.scss";
import Spells from "../Spells/Spells.jsx";
import PropTypes from "prop-types";

class SpellContainer extends Component {
  constructor() {
    super();
    this.state = {
      error: "",
      filtered: ""
    };
  }

  handleChange = (e) => {
    this.setState({
      filtered: e.target.value
    })
  }

  render() {
    let filteredSpells = this.props.spells.filter(spell => {
      return spell.spell.toLowerCase().indexOf(this.state.filtered) !== -1;
    });
    const displaySpells = filteredSpells.map(spell => (
      <Spells
        key={spell.id}
        spell={spell.spell}
        type={spell.type}
        effect={spell.effect}
      />
    ));
    return (
      <>
        <article>
          <h2 className="title">Harry Potter Spells</h2>
          <input className="search__input" type="text" value={this.state.filtered} onChange={this.handleChange} placeholder="Search" />
        </article>
        <section className="spells">{displaySpells}</section>
      </>
    );
  }
}

const mapStateToProps = state => ({
  spells: state.spells
});

// const mapDispatchToProps = dispatch => ({
//   setSpells: spells => dispatch(setSpells(spells))
// });

SpellContainer.propTypes = {
  spells: PropTypes.array,
  setSpells: PropTypes.func
};

export default connect(mapStateToProps)(SpellContainer);
