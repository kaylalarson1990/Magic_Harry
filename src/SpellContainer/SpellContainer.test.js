import { SpellContainer, mapStateToProps } from "./SpellContainer.jsx";
import { shallow } from "enzyme";
import React from "react";

describe(SpellContainer, () => {
  let wrapper, mockData, mockSubmit;
  beforeEach(() => {
    mockData = [
      {
        id: "1",
        spell: "Accio",
        type: "Sleeping spell",
        effect: "Puts to sleep on command"
      }
    ];

    mockSubmit = jest.fn();

    wrapper = shallow(<SpellContainer spells={mockData} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call setState on filtered", () => {
    const mockEvent = {
      target: {
        value: "Kayla"
      }
    };

    const expected = {
      error: "",
      filtered: "Kayla",
      num: 30
    };

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state()).toEqual(expected);
  });

  it('should be able to click "show more" button to see more data', () => {
    wrapper.instance().handleClick = jest.fn();
    wrapper.find(".showMoreBtn").simulate("click");
    expect(wrapper.instance().handleClick).toHaveBeenCalled();
  });
});

describe("mapStateToProps", () => {
  it("should return an object with a spell array", () => {
    const mockState = {
      spells: "Accio",
      type: "SET_SPELLS"
    };
    const expected = {
      spells: "Accio"
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
