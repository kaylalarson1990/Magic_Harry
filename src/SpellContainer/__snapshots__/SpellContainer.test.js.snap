// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpellContainer should match snapshot 1`] = `
<main>
  <article>
    <h2
      className="title"
    >
      Harry Potter Spells
    </h2>
    <input
      className="search__input"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </article>
  <section
    className="spells"
  >
    <Card
      info={
        Object {
          "effect": "Puts to sleep on command",
          "id": "1",
          "spell": "Accio",
          "type": "Sleeping spell",
        }
      }
    />
  </section>
  <div
    className="show-more"
  >
    <button
      className="showMoreBtn"
      onClick={[Function]}
    >
      Show More
    </button>
  </div>
</main>
`;
